File: 2274-keep-multiplying-found-values-by-two/keep-multiplying-found-values-by-two.java
class Solution {
    public int findFinalValue(int[] nums, int original) {
        boolean[] dp = new boolean[1001];
        for (int i : nums)
            dp[i] = true;
        while (original < 1001 && dp[original])
            original <<= 1;
        return original;
    }
}

File: 2274-keep-multiplying-found-values-by-two/keep-multiplying-found-values-by-two.cpp
class Solution {
public:
    int findFinalValue(vector<int>& nums, int original) {
        set<int> st;
        for(int i: nums)st.insert(i);
        while(st.find(original)!=st.end()){
            original=original*2;
        }
        return original;
    }
};